streaming/src/main/scala/io/gearpump/streaming/task/TaskActor.scala

```diff
@@ -242,11 +242,21 @@ class TaskActor(
       receiveMessage(inputMessage, sender)
     case upstream@ UpstreamMinClock(upstreamClock) =>
       this.upstreamMinClock = upstreamClock
-      val update = UpdateClock(taskId, minClock)
+      val latestMinClock = minClock
+      val update = UpdateClock(taskId, latestMinClock)
       context.system.scheduler.scheduleOnce(CLOCK_REPORT_INTERVAL) {
         appMaster ! update
       }
 
+      // check whether current task is dead.
+      if (latestMinClock > life.death) {
```

Comment:
When sending unRegister, it is not actually dead. 
It will only be dead when the AppMaster send back a StopTask.

During this transition time, we still need the clock to progress.